src/Persistence/Mapping/Driver/ColocatedMappingDriver.php

```diff
@@ -217,4 +220,24 @@ private function pathNameIterator(iterable $filesIterator): Generator
             yield $file->getPathname();
         }
     }
+
+    /**
+     * @param iterable<TKey, T> $iterable1
+     * @param iterable<TKey, T> $iterable2
+     *
+     * @return Generator<TKey, T>
+     *
+     * @template TKey
+     * @template T
+     */
+    private function mergeIterables(iterable $iterable1, iterable $iterable2): Generator
+    {
+        foreach ($iterable1 as $key => $item) {
+            yield $key => $item;
+        }
```

Comment:
I think this can be rewritten as

```suggestion
        yield from $iterable1;
```

Same below.